### DISCUSSION TITLE
Visualizing HMM fit on time series data

### DISCUSSION THREAD

**Discussion by mcmudge** [atsa-es/fish550-2023#36] [General]
After fitting some models and trying to evaluate the different states and covariates, it would be really nice to be able to visualize the output. I have some code for plotting the individual states, but was wondering if anyone has advice for overlaying these state plots onto the actual time series. It doesn't seem like the depmixS4 package has a lot of built in visual tools. 

# code to plot 3 state model with covariate effects on state parameters
```{r}
# plot posterior state sequence for the 3-state model
state1 <- plot(ts(posterior(fitmod3_3, type="smoothing")[,1], # [] state #
        start=c(1998)))
state2 <- plot(ts(posterior(fitmod3_3, type="smoothing")[,2], # [] state #
        start=c(1998)))
state3 <- plot(ts(posterior(fitmod3_3, type="smoothing")[,3], # [] state #
        start=c(1998)))

state1
state2
state3
```

**Comment by ericward-noaa** [comment]
Yeah, unfortunately the depmixS4 package doesn't play well with ggplot etc.

The first column of posterior(fit) will give you the viterbi states, which
represents the most probably sequence of states. I think you can use this
-- along with the means / intercepts for each of the time series to create
predicted values for each time series. Then I'd maybe bind them all into a
single dataframe, and use ggplot to show them all (using facetwrap() with
facets = variables?)






On Fri, May 5, 2023 at 9:36 PM Miranda Mudge ***@***.***>
wrote:

> After fitting some models and trying to evaluate the different states and
> covariates, it would be really nice to be able to visualize the output. I
> have some code for plotting the individual states, but was wondering if
> anyone has advice for overlaying these state plots onto the actual time
> series. It doesn't seem like the depmixS4 package has a lot of built in
> visual tools.
> code to plot 3 state model with covariate effects on state parameters
>
> # plot posterior state sequence for the 3-state model
> state1 <- plot(ts(posterior(fitmod3_3, type="smoothing")[,1], # [] state #
>         start=c(1998)))
> state2 <- plot(ts(posterior(fitmod3_3, type="smoothing")[,2], # [] state #
>         start=c(1998)))
> state3 <- plot(ts(posterior(fitmod3_3, type="smoothing")[,3], # [] state #
>         start=c(1998)))
>
> state1
> state2
> state3
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/atsa-es/fish550-2023/discussions/36>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABGQEZDQO5IMHYY6XPXNVU3XEVJC7ANCNFSM6AAAAAAXXQJ2BE>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 

*Eric Ward* (he/him)

Northwest Fisheries Science Center

NOAA Fisheries | U.S. Department of Commerce

Mobile: 206-650-7401
https://faculty.washington.edu/warde/
